README.md
Watch recaps, extended highlights, live and archived full games* with NHL TV
*Requires a valid NHL TV account to view full Live and Archived games. Blackout and other restrictions apply, including national blackouts.


changelog.txt
2016.11.15
- Added "No Preference" to CDN setting
- Added goal description to scoring notifications (goal scorer and assists, if any, appear in notification now)

2016.11.11.2
- Fixed CDN bug that could potentionally cause a too many sign-on restriction

2016.11.11
- Added a CDN option in settings (Under Login)

2016.10.18
- Fixed bug introduced in last version

2016.10.17
- Escape special characters in username and password (thanks Rel1k for pointing this out)

2016.10.13
- Fixed scoring notification error
- Updated login url

2016.7.27
- Fixed bug with Full Game Archive (Winter Classics didn't work)
- Goto date now employs select dialogs instead of text input
- Updated game thumbnails to incorporate latest team logos

2016.4.13
- Fixed bug

2016.4.12
- Fixed error when no games were available for date

2016.4.1
- Fixed debug code in Favorite Team' Game Today

2016.3.31
- Fixed some bugs with Favorite Team's Game Today
- Added favorite teams logo as icon for Favorite Teams list items
- Replaced "Favorite Team's..." with teams actual city in main menu
- Re-wrote favorite team code so favorite team variables are stored (less calls to the web)

2016.3.30
- Added Favorite Team's Game Today item that launches your favorite team's game and the team's feed. You can then add this as a favorite to make it easy to start your team's game and have it autopick your team's feed. (Thanks to Str8AWay for adding this feature!)
- Added setting to disable Game Preview Info (Today's Game fanart, game preview), which should help speed up load times for the Today's Games section
- Removed PIL requirement
- Added Score to Game Update notifications

2016.3.11
- Change today roll over to occur at 3 AM EST
- Added Game updates to scoring notifications. This displays a notification when a game starts or period starts (including OT & SO starts)
- To prevent any spoilers if the user is watching a game the "All games have finished notification" won't display now
- Fixed Featured Videos not playing when best available stream quality was selected
- Support free game of the day *Requires a nhl.com account (game will be highlighted in green)

2016.3.1
- Fix for scoring notifications not shutting down when all live games have finished
- Fix for Todays games rolling to the next day (est) when late games are still live

2016.2.24
- Fix for no spoilers in scoring notifications
- Fix for other bugs in scoring notificications
- User selected view mode is now saved and used throughout the addon
- Renamed NHL Videos to Featured Videos
- Added description info for the date list item (play all highlights item)

2016.2.19
- Fix for Todays Games saved to favorites
- Added new artwork
- Fix for scoring notification error
- Added game preview paragraph to todays games

2016.2.16
- Brought back scoring notifications
- Added Favorite Team Recent Games option (past 30 days)
- Added ability to view all recaps, extended highlights for a single day. Simply select the date list item and an option will pop on the screen to play recaps or extended highlights

2016.2.14
- Fixed Recap and Extended Highlights stream quality issue
- Added station identification to stream selection list items when applicable

2016.2.13
- Added No spoilers setting for archive games only
- Fix the way session keys are retrieved and stored
- Label Free game of the day
- Added Best Available for Stream Quality settings

2016.2.11
- Yet Another attempt to fix usage errors
- Added yesterday's games selection from main menu
- Added NHL Videos section
- Added No spoilers setting for live games only
- Added video info for archive games
- Switched default display to media info

2016.2.9
- Another attempt to fix usage errors

2016.2.6
- Fix for Usage errors
- Changed fanart
- New Color scheme for game types
- Code clean up

2016.2.3
- beta testing for nhl.tv overhaul

2016.1.29
- Added support for addition camera angles (i.e. Rogers Cams) *most games don't offer these*
- Added 5000kbps support for full and condensed archived games
- Convert local date to eastern when calling score notifications
- Fixed bug that caused score notifications to crash if no games were scheduled for the day
- Added audio and video stream info for display purposes

2016.1.20
- Favorite team color uses teams dominant color for highlighting
- Fixed favorite team bugs for certain teams
- Set default view mode for quick picks to media info 1
- Added previous and next icons

2016.1.16
*** Big thanks to mcruise40 for adding the following features ***
- Added ability to choose team naming style
- Added ability to choose time format
- Added ability to hide spoilers only for the favorite team
- favorite team will be in red in order to stick out even more :-)
- if no spoilers set to true, games decided in OT or SO will be shown as FINAL instead of FINAL (OT) or FINAL (SO)

2015.12.30
- Added team name to the two New York teams to distinguish them in the list
- Added ability to choose favorite team, so that there list title will be in bold in order to stick out more
- Bumped python version
- Added source to addon.xml

2015.10.29
- Minor Bug fix

2015.10.28

resources/lib/logout.py
import xbmc, xbmcplugin, xbmcgui, xbmcaddon
import urllib, urllib2
import cookielib
import os


ADDON = xbmcaddon.Addon(id='plugin.video.nhlgcl')
ADDON_PATH_PROFILE = xbmc.translatePath(ADDON.getAddonInfo('profile'))
UA_PC = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.97 Safari/537.36'
ROOTDIR = xbmcaddon.Addon(id='plugin.video.nhlgcl').getAddonInfo('path')

#Images
ICON = ROOTDIR+"/icon.png"

#Delete sessionKey
try:
    fname = os.path.join(ADDON_PATH_PROFILE, 'sessionKey.txt')
    os.remove(fname)
except:
    pass

cj = cookielib.LWPCookieJar(os.path.join(ADDON_PATH_PROFILE, 'cookies.lwp'))   
try:  
    cj.load(os.path.join(ADDON_PATH_PROFILE, 'cookies.lwp'),ignore_discard=True)
except:
    pass

opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(cj))                
url = 'https://account.nhl.com/ui/rest/logout'

req = urllib2.Request(url, data='',
      headers={"Accept": "*/*",
                "Accept-Encoding": "gzip, deflate",
                "Accept-Language": "en-US,en;q=0.8",
                "Content-Type": "application/x-www-form-urlencoded",                            
                "Origin": "https://account.nhl.com/ui/SignOut?lang=en",
                "Connection": "close",
                "User-Agent": UA_PC})

response = opener.open(req)              
user_data = response.read()
response.close()

#clear session cookies since they're no longer valid
cj.clear()
cj.save(ignore_discard=True);

dialog = xbmcgui.Dialog() 
title = "NHL TV" 
dialog.notification(title, 'Logout successfully completed', ICON, 5000, False) 





resources/lib/globals.py
# coding=utf-8
import sys
import xbmc, xbmcplugin, xbmcgui, xbmcaddon
import re, os, time
import calendar
import pytz
import urllib, urllib2
import json
import cookielib
import time
from bs4 import BeautifulSoup 
from datetime import date, datetime, timedelta
from urllib2 import URLError, HTTPError
#from PIL import Image
from cStringIO import StringIO

addon_handle = int(sys.argv[1])

#Addon Info
ADDON = xbmcaddon.Addon()
ADDON_ID = ADDON.getAddonInfo('id')
ADDON_VERSION = ADDON.getAddonInfo('version')
ADDON_PATH = xbmc.translatePath(ADDON.getAddonInfo('path'))
ADDON_PATH_PROFILE = xbmc.translatePath(ADDON.getAddonInfo('profile'))
XBMC_VERSION = float(re.findall(r'\d{2}\.\d{1}', xbmc.getInfoLabel("System.BuildVersion"))[0])
LOCAL_STRING = ADDON.getLocalizedString

#Settings
settings = xbmcaddon.Addon(id='plugin.video.nhlgcl')
USERNAME = json.dumps(str(settings.getSetting(id="username")))
PASSWORD = json.dumps(str(settings.getSetting(id="password")))
ROGERS_SUBSCRIBER = str(settings.getSetting(id="rogers"))
QUALITY = str(settings.getSetting(id="quality"))
NO_SPOILERS = settings.getSetting(id="no_spoilers")
FAV_TEAM = str(settings.getSetting(id="fav_team"))
TEAM_NAMES = settings.getSetting(id="team_names")
TIME_FORMAT = settings.getSetting(id="time_format")
VIEW_MODE = settings.getSetting(id='view_mode')
PREVIEW_INFO = str(settings.getSetting(id='game_preview_info'))

#Colors
SCORE_COLOR = 'FF00B7EB'
GAMETIME_COLOR = 'FFFFFF66'
#FAV_TEAM_COLOR = 'FFFF0000'
FREE_GAME_COLOR = 'FF43CD80'

#Game Time Colors
UPCOMING = 'FFD2D2D2'
LIVE = 'FFF69E20'
CRITICAL ='FFD10D0D'
FINAL = 'FF666666'
FREE = 'FF43CD80'

#Localization
local_string = xbmcaddon.Addon(id='plugin.video.nhlgcl').getLocalizedString
ROOTDIR = xbmcaddon.Addon(id='plugin.video.nhlgcl').getAddonInfo('path')

#Images
ICON = ROOTDIR+"/icon.png"
FANART = ROOTDIR+"/fanart.jpg"
#PREV_ICON = ROOTDIR+"/resources/images/prev.png"
#NEXT_ICON = ROOTDIR+"/resources/images/next.png"
PREV_ICON = ROOTDIR+"/icon.png"
NEXT_ICON = ROOTDIR+"/icon.png"

MASTER_FILE_TYPE = 'master_tablet60.m3u8'

#----------
#IPAD
#playstation
#WEB_MEDIAPLAYER
#----------
PLATFORM = "IPAD"

#User Agents
UA_GCL = 'NHL1415/5.0925 CFNetwork/711.4.6 Darwin/14.0.0'
UA_IPHONE = 'Mozilla/5.0 (iPhone; CPU iPhone OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/12H143 iphone nhl 5.0925'
UA_IPAD = 'Mozilla/5.0 (iPad; CPU OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/12H143 ipad nhl 5.0925'
UA_NHL = 'NHL/2542 CFNetwork/758.2.8 Darwin/15.0.0'
UA_PC = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.97 Safari/537.36'
UA_PS4 = 'PS4Application libhttp/1.000 (PS4) libhttp/3.15 (PlayStation 4)'

#Playlists
RECAP_PLAYLIST = xbmc.PlayList(0)
EXTENDED_PLAYLIST = xbmc.PlayList(1)


def find(source,start_str,end_str):    
    start = source.find(start_str)
    end = source.find(end_str,start+len(start_str))

    if start != -1:        
        return source[start+len(start_str):end]
    else:
        return ''

def getGameIcon(home,away):
    #Check if game image already exists
    image_path = ROOTDIR+'/resources/images/'+away+'vs'+home+'.png'
    file_name = os.path.join(image_path)
    if not os.path.isfile(file_name):
        image_path = ICON
    '''
    try:
        createGameIcon(home,away,image_path)
    except:
        pass
    '''

    return image_path

def createGameIcon(home,away,image_path):    
    bg = Image.new('RGB', (400,225), (255,255,255))    
    img_file = urllib.urlopen('http://nhl.bamcontent.com/images/logos/132x132/'+home.lower()+'.png ')
    im = StringIO(img_file.read())
    home_image = Image.open(im)

    img_file = urllib.urlopen('http://nhl.bamcontent.com/images/logos/132x132/'+away.lower()+'.png ')    
    im = StringIO(img_file.read())
    away_image = Image.open(im)

    bg.paste(away_image, (40,46), away_image)
    bg.paste(home_image, (228,46), home_image)
    bg.save(image_path)        
    

def colorString(string, color):
    return '[COLOR='+color+']'+string+'[/COLOR]'


def stringToDate(string, date_format):
    try:
        date = datetime.strptime(str(string), date_format)
    except TypeError:
        date = datetime(*(time.strptime(str(string), date_format)[0:6]))                

    return date


def easternToLocal(eastern_time):
    utc = pytz.utc
    eastern = pytz.timezone('US/Eastern')    
    eastern_time = eastern.localize(eastern_time)
    # Convert it from Eastern to UTC
    utc_time = eastern_time.astimezone(utc)
    timestamp = calendar.timegm(utc_time.timetuple())
    local_dt = datetime.fromtimestamp(timestamp)
    # Convert it from UTC to local time
    assert utc_time.resolution >= timedelta(microseconds=1)
    return local_dt.replace(microsecond=utc_time.microsecond)

def UTCToLocal(utc_dt):
    # get integer timestamp to avoid precision lost
    timestamp = calendar.timegm(utc_dt.timetuple())
    local_dt = datetime.fromtimestamp(timestamp)
    assert utc_dt.resolution >= timedelta(microseconds=1)
    return local_dt.replace(microsecond=utc_dt.microsecond)


def localToEastern():    
    eastern = pytz.timezone('US/Eastern')    
    local_to_utc = datetime.now(pytz.timezone('UTC'))  

    eastern_hour = local_to_utc.astimezone(eastern).strftime('%H')    
    eastern_date = local_to_utc.astimezone(eastern)
    #Don't switch to the current day until 4:01 AM est
    if int(eastern_hour) < 3:
        eastern_date = eastern_date - timedelta(days=1)  

    local_to_eastern = eastern_date.strftime('%Y-%m-%d')
    return local_to_eastern

def easternToUTC(eastern_time):    
    utc = pytz.utc
    eastern = pytz.timezone('US/Eastern')    
    eastern_time = eastern.localize(eastern_time)
    # Convert it from Eastern to UTC
    utc_time = eastern_time.astimezone(utc)
    return utc_time



def get_params():
    param=[]
    paramstring=sys.argv[2]
    if len(paramstring)>=2:
            params=sys.argv[2]
            cleanedparams=params.replace('?','')
            if (params[len(params)-1]=='/'):
                    params=params[0:len(params)-2]
            pairsofparams=cleanedparams.split('&')
            param={}
            for i in range(len(pairsofparams)):
                    splitparams={}
                    splitparams=pairsofparams[i].split('=')
                    if (len(splitparams))==2:
                            param[splitparams[0]]=splitparams[1]
                            
    return param




def addStream(name,link_url,title,game_id,epg,icon=None,fanart=None,info=None,video_info=None,audio_info=None,teams_stream=None,stream_date=None):
    ok=True
    u=sys.argv[0]+"?url="+urllib.quote_plus(link_url)+"&mode="+str(104)+"&name="+urllib.quote_plus(name)+"&game_id="+urllib.quote_plus(str(game_id))+"&epg="+urllib.quote_plus(str(epg))+"&teams_stream="+urllib.quote_plus(str(teams_stream))+"&stream_date="+urllib.quote_plus(str(stream_date))
    
    if icon != None:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon) 
    else:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
    
    if fanart != None:
        liz.setProperty('fanart_image', fanart)
    else:
        liz.setProperty('fanart_image', FANART)

    liz.setProperty("IsPlayable", "true")
    liz.setInfo( type="Video", infoLabels={ "Title": title } )
    if info != None:
        liz.setInfo( type="Video", infoLabels=info)
    if video_info != None:
        liz.addStreamInfo('video', video_info)
    if audio_info != None:
        liz.addStreamInfo('audio', audio_info)

    ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=False)
    xbmcplugin.setContent(addon_handle, 'episodes')
    
    return ok


def addFavToday(name,title,icon,fanart=None):
    info = {'plot':'','tvshowtitle':'NHL','title':name,'originaltitle':name,'aired':'','genre':'Sports'}
    audio_info, video_info = getAudioVideoInfo()
    ok=True    
    url = sys.argv[0] + '?url=/favteamCurrent&mode=510'
    if icon != None:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon) 
    else:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
    
    if fanart != None:
        liz.setProperty('fanart_image', fanart)
    else:
        liz.setProperty('fanart_image', FANART)

    liz.setProperty("IsPlayable", "true")
    liz.setInfo( type="Video", infoLabels={ "Title": title } )
    if info != None:
        liz.setInfo( type="Video", infoLabels=info)
    if video_info != None:
        liz.addStreamInfo('video', video_info)
    if audio_info != None:
        liz.addStreamInfo('audio', audio_info)

    ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=url,listitem=liz,isFolder=False)
    xbmcplugin.setContent(addon_handle, 'episodes')
    
    return ok


def addLink(name,url,title,iconimage,info=None,video_info=None,audio_info=None,fanart=None):
    ok=True
    liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)    
    liz.setProperty("IsPlayable", "true")
    liz.setInfo( type="Video", infoLabels={ "Title": title } )
    liz.setProperty('fanart_image', FANART)
    if iconimage != None:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
    else:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 

    if info != None:
        liz.setInfo( type="Video", infoLabels=info)
    if video_info != None:
        liz.addStreamInfo('video', video_info)
    if audio_info != None:
        liz.addStreamInfo('audio', audio_info)

    if fanart != None:
        liz.setProperty('fanart_image', fanart)
    else:
        liz.setProperty('fanart_image', FANART)

    ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=url,listitem=liz)
    xbmcplugin.setContent(addon_handle, 'episodes')
    return ok




def addDir(name,url,mode,iconimage,fanart=None,game_day=None):       
    ok=True    
    
    #Set day to today if none given
    #game_day = time.strftime("%Y-%m-%d")
    #game_day = localToEastern()
    #game_day = '2016-01-27'

    u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)+"&icon="+urllib.quote_plus(iconimage)
    if game_day != None:
        u = u+"&game_day="+urllib.quote_plus(game_day)

    if iconimage != None:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
    else:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 

    liz.setInfo( type="Video", infoLabels={ "Title": name } )

    if fanart != None:
        liz.setProperty('fanart_image', fanart)
    else:
        liz.setProperty('fanart_image', FANART)


    ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=True)    
    xbmcplugin.setContent(int(sys.argv[1]), 'episodes')
    return ok



def addPlaylist(name,game_day,url,mode,iconimage,fanart=None):       
    ok=True
    u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)+"&icon="+urllib.quote_plus(iconimage)

    if iconimage != None:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
    else:
        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 

    liz.setInfo( type="Video", infoLabels={ "Title": name } )

    if fanart != None:
        liz.setProperty('fanart_image', fanart)
    else:
        liz.setProperty('fanart_image', FANART)


    info = {'plot':'Watch all the days highlights for '+game_day.strftime("%m/%d/%Y"),'tvshowtitle':'NHL','title':name,'originaltitle':name,'aired':game_day.strftime("%Y-%m-%d"),'genre':'Sports'}
    audio_info, video_info = getAudioVideoInfo()

    if info != None:
        liz.setInfo( type="Video", infoLabels=info)
    if video_info != None:
        liz.addStreamInfo('video', video_info)
    if audio_info != None:
        liz.addStreamInfo('audio', audio_info)


    ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=False)    
    #xbmcplugin.setContent(int(sys.argv[1]), 'episodes')
    return ok


def scoreUpdates():
    #s = ScoreThread()
    t = threading.Thread(target = scoringUpdates)
    t.start() 



def getFavTeamId():
    url = 'http://statsapi.web.nhl.com/api/v1/teams/'
    req = urllib2.Request(url)   
    req.add_header('User-Agent', UA_IPAD)
    response = urllib2.urlopen(req)    
    json_source = json.load(response)                           
    response.close()

    fav_team_id = "0"    
    for team in json_source['teams']:
        if FAV_TEAM in team['name'].encode('utf-8'):
            fav_team_id = str(team['id'])
            break
    
    return fav_team_id


def getThumbnails():        
    try:
        from PIL import Image
    except:
        try:
            from pil import Image
        except:
            xbmc.log("PIL not available")
            sys.exit()
    
    url = 'http://statsapi.web.nhl.com/api/v1/teams/'
    req = urllib2.Request(url)   
    req.add_header('User-Agent', UA_IPAD)
    response = urllib2.urlopen(req)    
    json_source = json.load(response)                           
    response.close()
    
    team_list = []
    for team in json_source['teams']:
        team_list.append(team['abbreviation'].lower())

    img_url = 'http://nhl.bamcontent.com/images/logos/132x132/'
    for home_team in team_list:
        for away_team in team_list:
            if home_team != away_team:
                image_path = ROOTDIR+'/resources/images/'+away_team+'vs'+home_team+'.png'
                bg = Image.new('RGB', (400,225), (255,255,255))    
                img_file = urllib.urlopen('http://nhl.bamcontent.com/images/logos/132x132/'+home_team+'.png ')
                im = StringIO(img_file.read())
                home_image = Image.open(im)

                img_file = urllib.urlopen('http://nhl.bamcontent.com/images/logos/132x132/'+away_team+'.png ')    
                im = StringIO(img_file.read())
                away_image = Image.open(im)

                bg.paste(away_image, (40,46), away_image)
                bg.paste(home_image, (228,46), home_image)
                bg.save(image_path)

        

def getFavTeamColor():
    url = 'http://nhl.bamcontent.com/data/config/nhl/teamColors.json'
    req = urllib2.Request(url)   
    req.add_header('User-Agent', UA_IPAD)
    response = urllib2.urlopen(req)    
    json_source = json.load(response)                           
    response.close()

    fav_team_color = ''
    fav_team_id = settings.getSetting(id="fav_team_id")
    for team in json_source['teams']:
        if fav_team_id == str(team['id']):
            #Pick the lightest color
            fav_team_color = str(team['colors']['foreground'])
            if fav_team_color < str(team['colors']['background']):
                fav_team_color = str(team['colors']['background'])
            if fav_team_color < str(team['colors']['highlight']):
                fav_team_color = str(team['colors']['highlight'])
            
            fav_team_color = fav_team_color.replace('#','FF')
            break

    return fav_team_color

    '''
    #Hex code taken from http://teamcolors.arc90.com/        
    team_colors = {"Anaheim":"FF91764B",
                "Arizona":"FF841F27",
                "Boston":"FFFFC422",
                "Buffalo":"FF002E62",
                "Calgary":"FFE03A3E",
                "Carolina":"FFE03A3E",
                "Chicago":"FFE3263A",
                "Colorado":"FF8B2942",
                "Columbus":"FF00285C",
                "Dallas":"FF006A4E",
                "Detroit":"FFEC1F26",
                "Edmonton":"FF003777",
                "Florida":"FFC8213F",
                "Los Angeles":"FFAFB7BA",
                "Minnesota":"FF025736",
                "Montréal":"FFBF2F38",
                "Nashville":"FFFDBB2F",
                "New Jersey":"FFE03A3E",
                "New York Islanders":"FF00529B",
                "New York Rangers":"FF0161AB",
                "Philadelphia":"FFF47940",
                "Pittsburgh":"FFD1BD80",
                "Ottawa":"FFE4173E",
                "San Jose":"FF05535D",
                "St. Louis":"FF0546A0",
                "Tampa Bay":"FF013E7D",
                "Toronto":"FF003777",
                "Vancouver":"FF07346F",
                "Washington":"FFCF132B",
                "Winnipeg":"FF002E62"}
    
    # Default to red
    #fav_team_color = "FFFF0000"                
    #try:
    print FAV_TEAM
    fav_team_color = team_colors[FAV_TEAM]
    print fav_team_color
    #except:
    #pass
    '''
    return  fav_team_color


def getFavTeamLogo():
    logo_url = ''
    
    url = 'http://statsapi.web.nhl.com/api/v1/teams'
    req = urllib2.Request(url)   
    req.add_header('User-Agent', UA_IPAD)
    response = urllib2.urlopen(req)    
    json_source = json.load(response)                           
    response.close()

    fav_team_abbr = ''
    for team in json_source['teams']:
        if FAV_TEAM in team['name'].encode('utf-8'):
            fav_team_abbr = str(team['abbreviation']).lower()
            break

    if fav_team_abbr != '':        
        logo_url = 'http://nhl.bamcontent.com/images/logos/250x250/'+fav_team_abbr+'.png'
        
    return logo_url


def getAudioVideoInfo():
    #SD (800 kbps)|SD (1600 kbps)|HD (3000 kbps)|HD (5000 kbps)
    if QUALITY == 'SD (800 kbps)':        
        video_info = { 'codec': 'h264', 'width' : 512, 'height' : 288, 'aspect' : 1.78 }        
    elif QUALITY == 'SD (1200 kbps)':
        video_info = { 'codec': 'h264', 'width' : 640, 'height' : 360, 'aspect' : 1.78 }        
    else:
        #elif QUALITY == 'HD (2500 kbps)' or QUALITY == 'HD (3500 kbps)' or QUALITY == 'HD (5000 kbps)':
        video_info = { 'codec': 'h264', 'width' : 1280, 'height' : 720, 'aspect' : 1.78 }        

    audio_info = { 'codec': 'aac', 'language': 'en', 'channels': 2 }
    return audio_info, video_info


def getConfigFile():
    '''
    GET http://lwsa.mlb.com/partner-config/config?company=sony-tri&type=nhl&productYear=2015&model=PS4&firmware=default&app_version=1_0 HTTP/1.0
    Host: lwsa.mlb.com
    User-Agent: PS4Application libhttp/1.000 (PS4) libhttp/3.15 (PlayStation 4)
    Connection: close
    '''
    url = 'http://lwsa.mlb.com/partner-config/config?company=sony-tri&type=nhl&productYear=2015&model=PS4&firmware=default&app_version=1_0'
    req = urllib2.Request(url)       
    req.add_header("Connection", "close")
    req.add_header("User-Agent", UA_PS4)

    response = urllib2.urlopen(req, '')
    json_source = json.load(response)   
    response.close()

def setViewMode():
    global VIEW_MODE
    window = xbmcgui.Window(xbmcgui.getCurrentWindowId())
    current_view_mode = str(window.getFocusId())
    if current_view_mode != VIEW_MODE and current_view_mode != "0":
        settings.setSetting(id='view_mode', value=current_view_mode) 
        VIEW_MODE = settings.getSetting(id='view_mode')

    getViewMode()
    
    
def getViewMode():
    xbmc.executebuiltin("Container.SetViewMode("+VIEW_MODE+")")


def getAuthCookie():
    authorization = ''    
    try:
        cj = cookielib.LWPCookieJar(os.path.join(ADDON_PATH_PROFILE, 'cookies.lwp'))     
        cj.load(os.path.join(ADDON_PATH_PROFILE, 'cookies.lwp'),ignore_discard=True)    

        #If authorization cookie is missing or stale, perform login    
        for cookie in cj:            
            if cookie.name == "Authorization" and not cookie.is_expired():            
                authorization = cookie.value 
    except:
        pass

    return authorization


#Refresh Fav team info if fav team changed
if FAV_TEAM != str(settings.getSetting(id="fav_team_name")):
    if FAV_TEAM == 'None':
        settings.setSetting(id="fav_team_name", value='')
        settings.setSetting(id="fav_team_id", value='')
        settings.setSetting(id="fav_team_color", value='')
        settings.setSetting(id="fav_team_logo", value='')
    else:
        settings.setSetting(id="fav_team_name", value=FAV_TEAM)
        settings.setSetting(id="fav_team_id", value=getFavTeamId())
        settings.setSetting(id="fav_team_color", value=getFavTeamColor())
        settings.setSetting(id="fav_team_logo", value=getFavTeamLogo())


FAV_TEAM_ID = settings.getSetting(id="fav_team_id")
FAV_TEAM_COLOR = settings.getSetting(id="fav_team_color")
FAV_TEAM_LOGO = settings.getSetting(id="fav_team_logo")
